dm/master/scheduler/scheduler_test.go

```diff
@@ -1224,3 +1226,41 @@ func (t *testScheduler) TestStartStopSource(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(workers, HasLen, 0)
 }
+
+func checkAllWorkersClosed(c *C, s *Scheduler, closed bool) {
+	for _, worker := range s.workers {
+		cli, ok := worker.cli.(*workerrpc.GRPCClient)
+		c.Assert(ok, IsTrue)
+		c.Assert(cli.Closed(), Equals, closed)
+	}
+}
+
+func (t *testScheduler) TestCloseAllWorkers(c *C) {
+	var (
+		logger = log.L()
+		s      = NewScheduler(&logger, config.Security{})
+		names  []string
+	)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	for i := 1; i < 4; i++ {
+		names = append(names, fmt.Sprintf("worker%d", i))
+	}
+
+	for i, name := range names {
+		info := ha.NewWorkerInfo(name, fmt.Sprintf("127.0.0.1:%d", 50801+i))
```

Comment:
How did we successful dial the port if we didn't start worker there 🤔

Reply:
seems `conn.GetState()` is IDLE, so it didn't notice the endpoint is inaccessible 🤔